src/components/track-tile/TrackTile.tsx

```diff
@@ -109,70 +218,65 @@ export const TrackTile = ({
-          isArtistPick={isArtistPick}
-          isUnlisted={isUnlisted}
-          showArtistPick={showArtistPick}
-        />
-        <TrackTileMetadata
-          artistName={artistName}
-          coSign={coSign}
-          coverArtSizes={coverArtSizes}
-          fadeIn={fadeIn}
-          goToArtistPage={goToArtistPage}
-          goToTrackPage={goToTrackPage}
-          id={id}
-          isLoaded={isLoaded}
-          isPlaying={isPlaying}
-          setArtworkLoaded={setArtworkLoaded}
-          showSkeleton={showSkeleton}
-          title={title}
-          user={user}
-        />
-        {coSign && (
-          <Animated.View style={fadeIn}>
-            <TrackTileCoSign coSign={coSign} />
-          </Animated.View>
-        )}
-        <TrackTileStats
-          fadeIn={fadeIn}
-          hidePlays={hidePlays}
-          id={id}
-          index={index}
-          isTrending={isTrending}
-          isUnlisted={isUnlisted}
-          listenCount={listenCount}
-          makeGoToFavoritesPage={makeGoToFavoritesPage}
-          makeGoToRepostsPage={makeGoToRepostsPage}
-          repostCount={repostCount}
-          saveCount={saveCount}
-          showRankIcon={showRankIcon}
-        />
+        <Animated.View style={fadeIn}>
+          <TrackTileTopRight
+            duration={duration}
+            isArtistPick={_artist_pick === track_id}
+            isUnlisted={is_unlisted}
+            showArtistPick={showArtistPick}
+          />
+          <TrackTileMetadata
+            artistName={name}
+            coSign={_co_sign}
+            coverArtSizes={_cover_art_sizes}
+            goToArtistPage={goToArtistPage}
+            goToTrackPage={goToTrackPage}
+            id={track_id}
+            isPlaying={uid === playingUid && isPlaying}
+            setArtworkLoaded={setArtworkLoaded}
+            title={title}
+            user={user}
+          />
+          {_co_sign && <TrackTileCoSign coSign={_co_sign} />}
+          <TrackTileStats
+            hidePlays={hidePlays}
+            index={index}
+            isTrending={isTrending}
+            isUnlisted={is_unlisted}
+            listenCount={play_count}
+            onPressFavorites={onPressFavorites}
+            onPressReposts={onPressReposts}
+            repostCount={repost_count}
+            saveCount={save_count}
+            showRankIcon={showRankIcon}
+          />
+        </Animated.View>
         <TrackTileBottomButtons
-          hasSaved={hasCurrentUserSaved}
-          hasReposted={hasCurrentUserReposted}
-          toggleRepost={onToggleRepost}
-          toggleSave={onToggleSave}
-          onShare={onClickShare}
-          onPressOverflow={onClickOverflowMenu}
+          hasReposted={has_current_user_reposted}
+          hasSaved={has_current_user_saved}
           isOwner={isOwner}
-          isUnlisted={isUnlisted}
           isShareHidden={hideShare}
+          isUnlisted={is_unlisted}
+          onPressOverflow={onPressOverflowMenu}
```

Comment:
would it make sense to define all these callbacks in TrackTlieBottomButtons?

Reply:
That makes a lot more sense!